src/query/storage/m3/storage.go

```diff
@@ -790,6 +790,12 @@ func (s *m3storage) Write(
 		if !exists {
 			err = fmt.Errorf("no configured cluster namespace for: retention=%s,"+
 				" resolution=%s", attrs.Retention.String(), attrs.Resolution.String())
+			break
+		}
+		if namespace.Options().ReadOnly() {
```

Comment:
```suggestion
		} else if namespace.Options().ReadOnly() {
```

Reply:
This is how I wrote it initially, but then changed it. 
I believe in `go` culture `GOTO` is preferred over `else`.